include/dem/disable_particle_contact.h

```diff
@@ -0,0 +1,239 @@
+/* ---------------------------------------------------------------------
+ *
+ * Copyright (C) 2019 - 2022 by the Lethe authors
+ *
+ * This file is part of the Lethe library
+ *
+ * The Lethe library is free software; you can use it, redistribute
+ * it, and/or modify it under the terms of the GNU Lesser General
+ * Public License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * The full text of the license can be found in the file LICENSE at
+ * the top level of the Lethe distribution.
+ *
+ * ---------------------------------------------------------------------
+ *
+ */
+
+#include <core/dem_properties.h>
+
+#include <dem/data_containers.h>
+
+#include <deal.II/base/tensor.h>
+
+#include <deal.II/distributed/tria.h>
+
+#include <deal.II/dofs/dof_handler.h>
+
+#include <deal.II/lac/la_parallel_vector.h>
+#include <deal.II/lac/la_parallel_vector.templates.h>
+
+#include <deal.II/particles/particle_handler.h>
+
+#include <vector>
+
+using namespace dealii;
+
+#ifndef lethe_disable_particle_contact_h
+#  define lethe_disable_particle_contact_h
+
+// Special template instance for this class.
+// Unsigned integer would have been a better choice, but is not working
+// with the Vector class (error because of calling of abs() function)
+template class LinearAlgebra::distributed::Vector<int>;
+
+template <int dim>
+class DisableParticleContact
+{
+public:
+  DisableParticleContact<dim>();
+
+  /**
+   * Mobility status flag used to identify the status at nodes and the status
+   * of the cell
+   * inactive: movement of particles is negligible in the cell, particles won't
+   *           be in the contact candidates list
+   * active:   movement of particles is negligible, but there's at least one
+   * neighbor cell that is flag mobile, meaning that particles need to be in
+   * contact candidates lists mobile:   movement of particles is significant or
+   * there is at least one neighbor cell that is mobile by criteria, particles
+   * need to be in contact candidates lists empty:    cell is empty, only useful
+   * for mobility at nodes
+   */
+  enum mobility_status
+  {
+    inactive,
+    active,
+    mobile,
+    empty,
+    n_mobility_status
```

Comment:
I use mobility_status straight as an unsigned int everywhere, I don't know why should I use enum class, can you explain why it would be better Bruno?

Reply:
Ok, I understand. Right now, the hole thing is based on the fact that the mobility status are unsigned integers, so I have to refactor that. Here is an example : 

> mobility_at_nodes[dof_index] =
                    std::max((int)mobility_status::active,  mobility_at_nodes[dof_index]);

Also it is useful for output since the value is already the value. 
I will change the vector of map thing to a unordered map, and most of it will have to change and I think it will be easier to change the num to enum class. I don't really understand how/why it is such a bad thing, but I will find out how to do what you ask :)

Reply:
I refactored the class and looked more into that and I think using integer helps a lot for the nodes value assignation without overwriting. Otherwise i would need to have many ifs. Per instance, this part will be way more complicated.

>               for (auto node_id : local_dof_indices) {                
                  // Possible cases: (nodes are initialized to inactive (0))
                  // node = max(active (1), inactive (0)) = active (1)
                  // node = max(active (1), empty (3))    = empty (3)
                  // node = max(active (1), mobile (2))   = mobile (2)
                  mobility_at_nodes[node_id] =  std::max((int)mobility_status::active,
                                                                                                mobility_at_nodes[node_id]);
                }

What do you prefer?